### ISSUE TITLE
Table with merged headers not wrapping text

### ISSUE THREAD

**Issue by Cory-Kramer** [sphinx-doc/sphinx#3288]
Subject: <what happen when you do on which document project>

### Problem
- When any column headers are merged, text will not wrap correctly in other column headers

#### Procedure to reproduce the problem
Use the following sphinx table to reproduce the issue
```
+----------+--------------------------------+------------------------+----------+----------+----------+----------+----------+
| Header 1 | Long Header 2 that should wrap | Common column Header 3 | Header 4 | Header 5 | Header 6 | Header 7 | Header 8 |
|          |                                +-----------+------------+          |          |          |          |          |
|          |                                | Header 3a | Header 3b  |          |          |          |          |          |
+==========+================================+===========+============+==========+==========+==========+==========+==========+
| Value 1  | Value 2 does actually wrap     | Value 3a  | Value 3b   | Value 4  | Value 5  | Value 6  | Value 7  | Value 8  |
+----------+--------------------------------+-----------+------------+----------+----------+----------+----------+----------+
```
Then build using the latex builder
```
sphinx-build -b latex
```

#### Error logs / results
[Rendered table](https://i.stack.imgur.com/pLlpA.png)

#### Expected results
The text in the header of column 2 does not wrap, and overflows into the neighboring columns

### Reproducible project / your project
See the following Stack Overflow post for working and non-working examples
https://stackoverflow.com/questions/41131302/creating-a-large-table-in-rst-with-nested-column-headers-and-getting-the-latex-r

### Environment info
- OS: Win
- Python version: 3.5
- Sphinx version: 1.4.9
- tex compiler (e.g. MiKTeX)

**Comment by tk0miya** [comment]
Same as #3042.
AFAIK, it is hard to adjust table size automatically.
So please give hints to your LaTeX engines manually.

Thanks,

**Comment by jfbu** [comment]
Reopening because there is an issue here with Sphinx markup `\multirow{2}{*}` which does not take into account if user employed ``.. tabularcolumns::`` directive. The `*` according to `multirow` documentation sets its parameter text to natural width. Perhaps this means it can not wrap (not looked at source code -- yep, that's it actually). Although `multirow` is old package it got to 2.0 last September and now has `=` additionally to `*`. In my testing using `=` fixes the example in this issue. 

It would also help fix issue probably that if user has set a `p{2cm}` column spec, then Sphinx uses `{2cm}` in place of the `{*}`.  And this would not require recent up-to-date TeX installation. In testing this example `p{2cm}` is no good but `p{3cm}` is.

**Comment by jfbu** [comment]
`multirow` LaTeX package being only 5015 bytes, we could distribute it with Sphinx and then use `{=}` rather than `{*}` (need to investigate). But one would need to know if current multirow would work ok with 2013 TeX Live for example. Quick look suggests yes.

**Comment by jfbu** [comment]
Finally, it is proposed at #3435 Sphinx's own macros as replacement for package `multirow`. With this PR your table renders (on US letter size paper) as:

![capture d ecran 2017-02-22 a 09 05 11](https://cloud.githubusercontent.com/assets/2589111/23202393/0c57cbf2-f8de-11e6-9907-924ffef85898.png)

You can always for delicate tables use ``.. tabularcolumns`` directive for specifying manually column widths. Replacing 1.5.3 milestone by 1.6, because PR #3435 is currently on master branch.

**Comment by jfbu** [comment]
Closing now because this is fixed for 1.6 release at 74b4650. Thanks for the report.
